File: quantitative/QuantLibrary/src/Model/Exotics/Heston_PDE.cpp
#include "Exotics/Heston_PDE.h"

void HestonPDE::coefficients(double S, double v, double& dS, double& dV,
                             double& dSS, double& dVV, double& dSV, double& rTerm) {
    dS  = params.r * S;
    dV  = params.kappa * (params.theta - v);
    dSS = 0.5 * v * S * S;
    dVV = 0.5 * params.sigma * params.sigma * v;
    dSV = params.rho * params.sigma * v * S;
    rTerm = -params.r;
}

File: quantitative/quant_platform/src/analytics-service/analytics_service/models/exotics/heston_pde.py
import heston_pde

params = heston_pde.HestonParams()
params.kappa = 2.0
params.theta = 0.04
params.sigma = 0.3
params.rho = -0.7
params.v0 = 0.04
params.r = 0.05

price = heston_pde.heston_barrier_option_price(
    S0=100, K=100, B=90, T=1.0,
    hp=params,
    Nx=100, Nv=50, Nt=100
)

print(f"Option price: {price:.4f}")
